crates/subspace-farmer/src/bin/subspace-farmer/commands/bench.rs

```diff
@@ -306,3 +306,202 @@ fn get_size(path: impl AsRef<Path>) -> std::io::Result<u64> {
+
+    let seeder_max_plot_size = 20 * 1024 * 1024 / PIECE_SIZE as u64; // 20M
+    let syncer_max_plot_size = 2 * 1024 * 1024 / PIECE_SIZE as u64; // 20M
+    let request_pieces_size = 20;
+
+    let seeder_base_directory = TempDir::new().unwrap();
+    let mut seeder_multiaddr: Multiaddr = "/ip4/127.0.0.1/tcp/40000".parse().unwrap();
+
+    let (mut seeder_archived_segments_sender, seeder_archived_segments_receiver) =
+        mpsc::channel(10);
+    let seeder_client =
+        BenchRpcClient::new(BENCH_FARMER_METADATA, seeder_archived_segments_receiver);
+
+    let object_mappings = tokio::task::spawn_blocking({
+        let path = seeder_base_directory.as_ref().join("object-mappings");
+
+        move || ObjectMappings::open_or_create(path)
+    })
+    .await
+    .unwrap()
+    .unwrap();
+
+    let base_path = seeder_base_directory.as_ref().to_owned();
+    let plot_factory = move |plot_index, public_key, max_piece_count| {
+        let base_path = base_path.join(format!("plot{plot_index}"));
+        Plot::open_or_create(base_path, public_key, max_piece_count)
+    };
+
+    let seeder_multi_farming = MultiFarming::new(
+        MultiFarmingOptions {
+            base_directory: seeder_base_directory.as_ref().to_owned(),
+            archiving_client: seeder_client.clone(),
+            farming_client: seeder_client.clone(),
+            object_mappings: object_mappings.clone(),
+            reward_address: PublicKey::default(),
+            bootstrap_nodes: vec![],
+            listen_on: vec![seeder_multiaddr.clone()],
+            dsn_sync: false,
+        },
+        seeder_max_plot_size,
+        seeder_max_plot_size,
+        plot_factory,
+        false,
+    )
+    .await
+    .unwrap();
+
+    {
+        let mut last_archived_block = LastArchivedBlock {
+            number: 0,
+            archived_progress: ArchivedBlockProgress::Partial(0),
+        };
+
+        for segment_index in 0..seeder_max_plot_size / PIECE_SIZE as u64 / 256 {
+            last_archived_block
+                .archived_progress
+                .set_partial(segment_index as u32 * 256 * PIECE_SIZE as u32);
+
+            let archived_segment = {
+                let root_block = RootBlock::V0 {
+                    segment_index,
+                    records_root: Sha256Hash::default(),
+                    prev_root_block_hash: Sha256Hash::default(),
+                    last_archived_block,
+                };
+
+                let mut pieces = FlatPieces::new(256);
+                rand::thread_rng().fill(pieces.as_mut());
+
+                let objects = std::iter::repeat_with(|| PieceObject::V0 {
+                    hash: rand::random(),
+                    offset: rand::random(),
+                })
+                .take(100)
+                .collect();
+
+                ArchivedSegment {
+                    root_block,
+                    pieces,
+                    object_mapping: vec![PieceObjectMapping { objects }],
+                }
```

Comment:
Not using archiver will probably strike later because pieces that are just randomly generated will fail validation once we add it, so while this might work for now, it will add extra friction for introduction of validation later down the road and this will have to be rewritten anyway.

Reply:
In that case, we would need to rewrite the bench command also.